### PR TITLE
SOLR-6443: Cleanup managedResourceTest

### PR DESCRIPTION
**Pull request by HoustonPutman**
Mainly remove the need for a SolrCore in the test.

https://issues.apache.org/jira/browse/SOLR-6443

### CODE REVIEW

File: solr/test-framework/src/java/org/apache/solr/cloud/AbstractZkTestCase.java [hunk 1]
```diff
@@ -62,39 +51,22 @@ public static void azt_beforeClass() throws Exception {
     System.setProperty("jetty.port", "0000");
     System.setProperty(ZOOKEEPER_FORCE_SYNC, "false");
     
-    zkServer.buildZooKeeper(SOLRHOME,
+    zkServer.buildZooKeeper(SolrTestCaseJ4.TEST_PATH(),
```
Comment (madrob) [inline solr/test-framework/src/java/org/apache/solr/cloud/AbstractZkTestCase.java:54]: I wouldn't inline this... at worst, copy `getSolrHome` from BaseDistributesSearchTestCase and use that. The problem is just like the comment up above says - if we're calling this from solrj or from a third party that uses our test-framework (and this is technically publicly available) then TEST_PATH won't work and they'll have no way to override/inject the correct value for their installation.

File: solr/core/src/test/org/apache/solr/rest/TestManagedResourceStorage.java [hunk 2]
```diff
@@ -116,7 +103,7 @@ private void doStorageTests(SolrResourceLoader loader, StorageIO storageIO) thro
     assertEquals("true", storedArgs.get("ignoreCase"));
     List<String> storedList = (List<String>)storedMap.get(ManagedResource.MANAGED_JSON_LIST_FIELD);
     assertNotNull(storedList);
-    assertTrue(storedList.size() == managedList.size());
+    assertEquals(storedList.size(), managedList.size());
```
Comment (madrob) [inline solr/core/src/test/org/apache/solr/rest/TestManagedResourceStorage.java:106]: nit: I think managedList is the expected value, swap the order of these arguments.

File: solr/core/src/test/org/apache/solr/rest/TestManagedResourceStorage.java [hunk 3]
```diff
@@ -36,28 +35,19 @@
...
   public void testZkBasedJsonStorage() throws Exception {
-    
-    // test using ZooKeeper
-    assertTrue("Not using ZooKeeper", h.getCoreContainer().isZooKeeperAware());
-    SolrResourceLoader loader = new SolrResourceLoader(Paths.get("./"));
     // Solr unit tests can only write to their working directory due to
     // a custom Java Security Manager installed in the test environment
-    NamedList<String> initArgs = new NamedList<>();
-    try {
+    try (SolrResourceLoader loader = new SolrResourceLoader(Paths.get("./"))) {
```
[the remark this answers is not in the document]
Comment (HoustonPutman) [reply solr/core/src/test/org/apache/solr/rest/TestManagedResourceStorage.java:47]: probably a temporary directory

File: solr/test-framework/src/java/org/apache/solr/cloud/AbstractZkTestCase.java (hunk 1 above)
Comment (HoustonPutman) [reply solr/test-framework/src/java/org/apache/solr/cloud/AbstractZkTestCase.java:54]: So `AbstractZkTestCase` is only used by `TestManagedResourceStorage`, which doesn't actually use the configs stored in Zookeeper, so we can go-ahead and just remove this 🙂

**Pull request closed without merging**